services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py

```diff
@@ -0,0 +1,130 @@
+""" Handles request to the viewers redirection entrypoints
+
+"""
+import logging
+from typing import Optional
+
+import aiohttp
+from aiohttp import web
+from aiohttp.client_exceptions import ClientError
+from pydantic import BaseModel, HttpUrl, ValidationError
+from pydantic.types import PositiveInt
+from yarl import URL
+
+from ..statics import INDEX_RESOURCE_NAME
+from ._core import (
+    MatchNotFoundError,
+    ValidationMixin,
+    ViewerInfo,
+    find_compatible_viewer,
+)
+from ._projects import acquire_project_with_viewer
+from ._users import UserInfo, acquire_user, ensure_authentication
+
+log = logging.getLogger(__name__)
+
+
+def create_redirect_response(
+    app: web.Application, page: str, **parameters
+) -> web.HTTPFound:
+    """
+    Returns a redirect response to the front-end with information on page and parameters embedded in the fragment.
+
+    For instance,
+        https://osparc.io/#/error?message=Sorry%2C%20I%20could%20not%20find%20this%20&status_code=404
+    results from
+            - page=error
+        and parameters
+            - message="Sorry, I could not find this"
+            - status_code=404
+    """
+    page = page.strip(" /")
+    # TODO: test that fragment queries are understood by front-end
+    # TODO: front end should create an error page and a view page
+    assert page in ("view", "error")  # nosec
```

Comment:
Can the assert be replaced with something that cannot be accidentally removed?

Reply:
those are indeed to be removed ... all asserts are removed in production. This is simply for defensive programming during testing to notice if static invariants change